packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -103,23 +90,28 @@ Future<List<Plugin>> findPlugins(
     logger: globals.logger,
     throwOnError: throwOnError,
   );
-  final Set<String> devDependencies;
-  if (!determineDevDependencies) {
-    devDependencies = <String>{};
-  } else {
-    devDependencies = await computeExclusiveDevDependencies(
-      pub,
-      logger: globals.logger,
-      project: project,
-    );
-  }
-  for (final Package package in packageConfig.packages) {
-    final Uri packageRoot = package.packageUriRoot.resolve('..');
+  final Map<String, Dependency> transitiveDependencies = computeTransitiveDependencies(
+    project,
+    packageConfig,
+    fs,
+  );
+  for (final String packageName in transitiveDependencies.keys) {
+    final Dependency dependency = transitiveDependencies[packageName]!;
```

Comment:
Ubernit: this variable declaration could be moved after the conditional below.

Reply:
Done